tests/UI/specs/PeriodSelector_spec.js

```diff
@@ -134,4 +134,33 @@ describe("PeriodSelector", function () {
 
         expect(await page.screenshotSelector(selector + ',#notificationContainer')).to.matchImage('invalid');
     });
+
+    it('should preselect previous "compare to custom range" on load', async function() {
```

Comment:
I like that you have covered this all with UI tests, however the difference between them all is just the `comparePeriodType` and the first one have `compareDates`.

I'm wondering if this slight differences warrant a full UI test, and they aren't something that can be tested with a unit/integration test? I'm not even sure if we support writing those kind of tests for Vue, but just a thought.

Reply:
I condensed the screenshots down to the custom range one, that one does the most "work" with filling the date inputs. Other values are verified only with a simple input value match to replace the screenshot to at cover the cases without much overhead.